e2e_testing/torchscript/histogram_binning_calibration.py

```diff
@@ -0,0 +1,190 @@
+        )
+        self.register_buffer(
+            "_bin_num_positives",
+            torch.empty([_num_interval], dtype=torch.float64).fill_(0.0),
+        )
+        self.register_buffer("_bin_ids", torch.arange(_num_interval))
+        self.positive_weight = torch.tensor([0.4])
+        self.bin_ctr_in_use_after = 0
+        self.bin_ctr_weight_value = 0.9995
+        self.oneminusbin_ctr_weight_value = 0.0005
+        self._iteration = 0
+
+    @export
+    @annotate_args([
+        None,
+        ([-1], torch.int32, True),
+        ([-1], torch.int32, True),
+        ([-1], torch.float32, True),
+    ])
+    def forward(self, segment_value, segment_lengths, logit):
+        origin_prediction = torch.sigmoid(
+            logit + torch.log(self.positive_weight))
+        dense_segment_value = torch.zeros(logit.numel(), dtype=torch.int32)
+        validoffsets = torch.gt(
+            segment_lengths[1:self._num_logits+1], segment_lengths[0:self._num_logits])
+        gathered_segment_values = (
+            segment_value[segment_lengths[0:self._num_logits].long()]+1).int()
+        dense_segment_value = torch.where(
+            validoffsets, gathered_segment_values, dense_segment_value)
+        zeros = torch.empty_like(
+            dense_segment_value, dtype=torch.int32).fill_(0)
+        isnotvalid = torch.gt(dense_segment_value, self._num_segments)
+        dense_segment_value = torch.where(
+            isnotvalid, zeros, dense_segment_value)
+        bin_ids_data = torch.ceil(origin_prediction/self.step)-1
+        bin_ids_data = bin_ids_data.long()
+        curr_segment_value = dense_segment_value * self._num_bins
+        bin_ids_data2 = bin_ids_data
+        bin_ids_data = bin_ids_data + curr_segment_value
+        curr_segment_value = self._bin_num_positives[bin_ids_data]
+        curr_bin_num_examples = self._bin_num_examples[bin_ids_data]
+        curr_segment_value = curr_segment_value / curr_bin_num_examples
+        curr_segment_value = curr_segment_value.float()
+        curr_segment_value = curr_segment_value * self.bin_ctr_weight_value + \
+            origin_prediction * self.oneminusbin_ctr_weight_value
+        isvalid = torch.gt(curr_bin_num_examples,
+                           self.bin_ctr_in_use_after)
+        calibrated_prediction_data = torch.where(
+            isvalid, curr_segment_value, origin_prediction.float())
+        return calibrated_prediction_data, bin_ids_data
+
+
+@register_test_case(module_factory=lambda: HistogramBinningCalibrationByFeature())
+def HBC_basic(module, tu: TestUtils):
+    segment_value, segment_lengths, logit = make_HBC_inputs()
+    module.forward(segment_value, segment_lengths, logit)
+
+# This is a specialized case to support the FBGEMM input size of 5000
+# https://github.com/pytorch/FBGEMM/blob/main/fbgemm_gpu/bench/histogram_binning_calibration_benchmark.py
+
+
+class HistogramBinningCalibrationByFeatureStaticSize(torch.nn.Module):

```

Comment:
Based on the prior discussion, it seems that the only reason to have the static shape test was a bug in IREE. I don't think that's a strong enough reason to keep a duplicated test in our test suite. So can you remove the static shape version of the test?

Reply:
I would recommend that downstreams that really care about this test running on IREE carry a local patch to make this statically shaped until the iree issues are resolved.

Reply:
Thanks for the feedback. Sounds like a good plan.